eyecite/models.py

```diff
@@ -114,7 +114,20 @@ def comparison_hash(self) -> int:
         if isinstance(self, CaseCitation) and self.groups["page"] is None:
             return id(self)
         else:
-            return hash((type(self), tuple(self.groups.items())))
+            # Groups contains the original reporter, use the corrected one instead
```

Comment:
Would it make more sense to implement this as an overriding method on the child `CaseCitation` class (objects of which I believe are guaranteed to have a `reporter` key in the `groups` attribute) instead?

Also looking at line 114 above, I feel like that `isinstance()` check should be replaced by simply moving that logic into the same (proposed) `comparison_hash()` override at the `CaseCitation` level. (I believe I added that code myself, so this is self-criticism! Just doesn't seem very polymorphic to me.)

Reply:
Yes you are totally right, it would be even better, I wasn't sure of the requirement on the `reporter` key being present if and only if the citation is a `CaseCitation`

I don't really get why we need to hash the `type`, as the groups should already be citation-specific (I guess).

Reply:
> I wasn't sure of the requirement on the `reporter` key being present if and only if the citation is a `CaseCitation`

Yeah, I wish there were some way we could statically enforce this in the `eyecite` code with `mypy` but I can't think of one. However, this test here in `reporters_db` should guarantee the presence of the `reporter` key in the regexes used to generate `CaseCitation` objects: https://github.com/freelawproject/reporters-db/blob/main/tests.py#L129

> I don't really get why we need to hash the `type`, as the groups should already be citation-specific (I guess).

Me neither, but I guess keep it to be safe?